### PR TITLE
complete sti builder tests

### PR DESCRIPTION
**Pull request by PI-Victor**
add missing tests for sti builder in `pkg/build/builder/sti_test.go`

### CODE REVIEW

File: pkg/build/builder/sti_test.go [hunk 1]
```diff
@@ -145,3 +153,65 @@ func TestCopyToVolumeList(t *testing.T) {
...
+			Name:  "OPENSHIFT_BUILD_SOURCE",
+			Value: "http://localhost/123",
+		},
+	}
+	mockBuild := makeBuild()
+	mockBuild.Name = "openshift-test-1-build"
+	mockBuild.Namespace = "openshift-demo"
+	resultedEnvList := buildEnvVars(mockBuild)
+	if !reflect.DeepEqual(expectedEnvList, resultedEnvList) {
+		t.Errorf("Expected EnvironmentList to match: %#v, got %#v", expectedEnvList, resultedEnvList)
```
Comment (PI-Victor) [inline pkg/build/builder/sti_test.go]:
```
sti_test.go:185: Expected EnvironmentList to match: 
api.EnvironmentList{
  api.EnvironmentSpec{
    Name:"HTTP_PROXY", 
    Value:"http://test/insecure:8080"
  }, api.EnvironmentSpec{
     Name:"HTTPS_PROXY", 
     Value:"https://test/secure:8443"
   }, api.EnvironmentSpec{
     Name:"OPENSHIFT_BUILD_NAME", 
     Value:"openshift-test-1-build"
    }, api.EnvironmentSpec{
     Name:"OPENSHIFT_BUILD_NAMESPACE", 
     Value:"openshift-demo"
    }, api.EnvironmentSpec{
     Name:"OPENSHIFT_BUILD_SOURCE", 
     Value:"http://localhost/123"
  }
}, 
got 
api.EnvironmentList{
  api.EnvironmentSpec{
    Name:"OPENSHIFT_BUILD_NAME", 
    Value:"openshift-test-1-build"
  }, api.EnvironmentSpec{
    Name:"OPENSHIFT_BUILD_NAMESPACE", 
    Value:"openshift-demo"
   }, api.EnvironmentSpec{
    Name:"OPENSHIFT_BUILD_SOURCE", 
    Value:"http://localhost/123"
    }, api.EnvironmentSpec{
    Name:"HTTPS_PROXY", 
    Value:"https://test/secure:8443"
   }, api.EnvironmentSpec{
    Name:"HTTP_PROXY", 
    Value:"http://test/insecure:8080"
  }
}
```

@php-coder i need a sharp eye, can you tell me why the comparison between these two structure, fails? i can't seem to put my finger on it

Comment (php-coder) [inline pkg/build/builder/sti_test.go]: @PI-Victor I suspect that the problem here is because of order of elements in the array.

**Comment by PI-Victor** [comment]
@bparees added some tests that were missing.
wanted to test also downloader.Download, but because gitFetchSource is isolated, i'd have the change the function definition all around, so i've decided to leave it as it is.

see: https://github.com/openshift/origin/blob/master/pkg/build/builder/sti.go#L299
and https://github.com/openshift/origin/blob/master/pkg/build/builder/source.go#L48

**Comment by bparees** [comment]
what do you mean "gitFetchSource is isolated"?

**Comment by PI-Victor** [comment]
i couldn't mock it, because it doesn't belong to any struct, so what after i did a bit of reading, i came to this http://stackoverflow.com/a/19168875/2146586

the problem with gitfetchsource is that it goes and tries to modify directories and whatnot and calls a few other functions as well.
is there some other approach to this?

**Comment by bparees** [comment]
@PI-Victor gotcha.  yeah i don't have a good solution for you.  long term gitFetchSource should probably be made part of an interface, but I agree it's not the right thing to spend time on now.

this PR lgtm, is it still WIP?

**Comment by PI-Victor** [comment]
@bparees s/struct/interface/ on my previous comment :)

we can merge, if all is good.

**Comment by bparees** [comment]
[merge]

File: pkg/build/builder/sti_test.go [hunk 2]
```diff
@@ -146,3 +154,66 @@ func TestCopyToVolumeList(t *testing.T) {
...
+	}
+	resultedProxyConf, err := scriptProxyConfig(newBuild)
+	if err != nil {
+		t.Errorf("An error occured while parsing the proxy config: %v", err)
+	}
+	if resultedProxyConf.HTTPProxy.Path != "/insecure" {
+		t.Errorf("Expected HTTP Proxy path to be /insecure, got: %v", resultedProxyConf.HTTPProxy.Path)
+	}
+	if resultedProxyConf.HTTPSProxy.Path != "/secure" {
+		t.Errorf("Expected HTTP Proxy path to be /secure, got: %v", resultedProxyConf.HTTPSProxy.Path)
```
Comment (php-coder) [inline pkg/build/builder/sti_test.go]: HTTP -> HTTPS?

File: pkg/build/builder/sti_test.go [hunk 3]
```diff
@@ -146,3 +154,66 @@ func TestCopyToVolumeList(t *testing.T) {
...
+							Value: "http://test/insecure",
+						}),
+					},
+				},
+			},
+		},
+	}
+	resultedProxyConf, err := scriptProxyConfig(newBuild)
+	if err != nil {
+		t.Errorf("An error occured while parsing the proxy config: %v", err)
```
Comment (php-coder) [inline pkg/build/builder/sti_test.go]: Consider using `t.Fatalf()` here.

File: pkg/build/builder/sti_test.go (hunk 2 above)
Comment (bparees) [inline pkg/build/builder/sti_test.go]: good catch, i'll yank the merge tag

Comment (PI-Victor) [inline pkg/build/builder/sti_test.go]: copy paste, will fix. thanks

**Comment by PI-Victor** [comment]
@bparees @php-coder nits fixed.

**Comment by bparees** [comment]
[merge]

**Pull request merged**